test/studies/adventOfCode/2022/day11/bradc/day11-par.chpl

```diff
@@ -0,0 +1,181 @@
+use IO, List, Barriers;
+
+config const numRounds = 20;
+
+// things that define a monkey
+class Monkey {
+  var id: int,                 // an integer ID
+      Items: [0..1] list(int, parSafe=true),  // two lists of its items
+      current = 0, next=1,     // which list is the current vs. next one
+      op: owned MathOp,            // the operator the monkey does
+      divisor: int,            // the divisor for its % operation check
+      targetMonkey: 2*int,     // the target monkeys it throws items to
+      numInspected: int;       // the count of how many inspections we've done
+}
+
+/*
+var m1 = new Monkey();
+var m2 = new Monkey();
+var m3 = new Monkey();
+var m4 = new Monkey();
+*/
+
+// An array of monkeys
+var Monkeys = readMonkeys();
+//writeln(Monkeys);
+
+
+// This tells whether a given monkey can proceed when it is out of items.
+// Initially, only monkey 0 can since nobody can throw items into its
+// list for this round, only for subsequent rounds.
+var canPause: sync int = 0;
```

Comment:
Isn't this name inverted? Should it be `canProceed`? At least the comment makes me think so.

Reply:
Ah, this is to save a monkey work when no more items will be thrown its way in a round. I think this variable could be name something else, but I'm not 100% sure what a good name would be.

Reply:
Yeah, I've struggled with the name as well.  It was originally called `canProceed` which is why the comment uses that phrase.  Then I realized that it's like "proceed to do nothing but wait until we're ready to set up for the next round, which led to the current name.  I'll mull it over some more.

Reply:
I like `canProceed`.  I read it as can proceed to setting up for the next round.

Reply:
I went back to the problem statement for inspiration and ended up going with `canFinishTurn`, at least for now.